packages/components/src/Chip/Chip.tsx

```diff
@@ -41,19 +42,21 @@ export const Chip = ({
     label,
     heading,
   );
+  const Tag = onClick ? "button" : "div";
```

Comment:
I realize that this might be out of scope of this PR, since it's only trying to re-revert the original `Chip` PR. However, I have some accessibility concerns with using `button` as the container for `Chip`.
[According to W3C](https://www.w3.org/TR/2011/WD-html5-20110525/the-button-element.html), the button can't have `div` elements as its descendants, only "phrasing content". However, right now this is not the case:
<img width="712" alt="image" src="https://github.com/user-attachments/assets/e0e59065-a023-47ea-8da0-8186bb5ecb4f">

I think the safer approach from the accessibility standpoint would be to always use `div role="button"` and change internal elements (prefix, content, suffix) to be buttons (if they don't have any descendants) or other `div`s, `span`s, `p`s etc.

I'm also not quite sure whether it's currently the case, but there's a chance that the parent component that would use `Chip` instances could also be a `button` element itself. In which case, my suggestion with using `div` as the container for `Chip` will also break accessibility, meaning that it will probably need to be a `<span role="button">`.

If you feel like this is not the best time and place to refactor this, I will totally understand. But at least I think this would be worth logging it somewhere as future improvement.

Reply:
that's a fair callout. divs as descendants isn't correct for a button.

I think this has changed quite a bit in the past year, and even if we did change `chipContent` and other internal elements into `span`s or something since we have `Chip.Prefix` open to be whatever, we cannot guarantee that whatever is provided will be semantically valid. 

it probably does make sense to have it be a div all the time. that said, unless it's as simple as making it a div and everything still works - I'd be hesitant to make that change with this PR. I assume there's a reason we want it to be a button sometimes.

>I'm also not quite sure whether it's currently the case, but there's a chance that the parent component that would use Chip instances could also be a button element itself. In which case, my suggestion with using div as the container for Chip will also break accessibility, meaning that it will probably need to be a <span role="button">.

I'd argue that's possible with every component. you could put any component inside a button if you wanted to, and it would become semantically invalid. I don't think we should worry about that when building components unless we reasonably expect people to put them inside a button. since Chip has an `onClick` already, I would say wrapping it in a button isn't a reasonable thing to do.